torch/testing/_internal/common_distributed.py

```diff
@@ -1456,12 +1475,18 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
 
 @contextmanager
 def _dynamo_dist_per_rank_init(
-    rank, world_size, backend="nccl", init_pg=True, fake_pg=False
+    rank, world_size, backend=None, init_pg=True, fake_pg=False
 ):
     # To avoid multiple inheritance from _dynamo.test_case.TestCase and MultiProcessTestCase,
     # Just manually implement the most important part of the dynamo behavior to reset/clear.
     if not fake_pg:
         torch.accelerator.set_device_index(rank)
+
+    if backend is None:
+        backend = c10d.distributed_c10d.Backend.default_device_backend_map.get(
```

Comment:
Please use `get_default_backend_for_device` instead; the latter one has a better error message.